beacon-chain/db/kv/state_summary.go

```diff
@@ -21,10 +25,20 @@ func (s *Store) SaveStateSummary(ctx context.Context, summary *ethpb.StateSummar
 func (s *Store) SaveStateSummaries(ctx context.Context, summaries []*ethpb.StateSummary) error {
 	ctx, span := trace.StartSpan(ctx, "BeaconDB.SaveStateSummaries")
 	defer span.End()
+	return s.SaveStateSummariesWithPendingBlocks(ctx, summaries, nil)
+}
+
+// SaveStateSummariesWithPendingBlocks saves state summary objects to the DB.
+func (s *Store) SaveStateSummariesWithPendingBlocks(ctx context.Context, summaries []*ethpb.StateSummary, blockCache func([32]byte) (interfaces.SignedBeaconBlock, error)) error {
+	ctx, span := trace.StartSpan(ctx, "BeaconDB.SaveStateSummariesWithPendingBlocks")
+	defer span.End()
 
 	// When we reach the state summary cache prune count,
 	// dump the cached state summaries to the DB.
 	if s.stateSummaryCache.len() >= stateSummaryCachePruneCount {
+		if err := s.ensureBlocksSaved(ctx, blockCache); err != nil {
```

Comment:
This function seems to ignore the blocks that are not in the cache and are not in the summaries. It will not return an error and the summaries will be saved down below without a corresponding block in DB

Reply:
See line 137 for the check that blocks are in the db.